config/initializers/aws.rb

```diff
@@ -1,7 +1,13 @@
+pattern =
+  ':client_class :http_response_status_code :time :retries :operation [:error_class :error_message]'
```

Comment:
> Ideally we'd do JSON so Cloudwatch could automatically parse logs from the AWS SDK, but it doesn't appear to be possible.

Could we trick it into printing JSON like this?

```suggestion
  '{"client_class":":client_class", "http_response":":http_response_status_code", "time": ":time", "retries": ":retries", "operation": ":operation", "error": "[:error_class :error_message]"}'
```

Reply:
hmmmmmm we could also just subclass and implement the JSON formatting ourselves? I wouldn't mind calling the private methods if we added tests for them 😬 

Or we could just try to access the properties of `response.context` like they do in their source:

https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-core/lib/aws-sdk-core/log/formatter.rb#L120

```
class JsonAwsLogFormatter
  def format(response)
    {
      client_class: response.context.client.class.name,
      http_response_status_code: response.context.http_response.status_code.to_s,
      # ... etc etc
    }.to_json
  end
end
```